### PR TITLE
Set the prepare_alpaca test split size as a fraction of the data set size

### PR DESCRIPTION
**Pull request by patrickhwood**
Enhancement to avoid https://github.com/Lightning-AI/lit-gpt/issues/260

Compute test split based on percentage of dataset size instead of from a fixed number of samples.

### CODE REVIEW

File: scripts/prepare_alpaca.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 DATA_FILE_NAME = "alpaca_data_cleaned_archive.json"
 DESTINATION_PATH = Path("data/alpaca")
 CHECKPOINT_DIR = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
-TEST_SPLIT_SIZE = 2000
+TEST_SPLIT_PERCENT = 10
```
Comment (carmocca) [inline scripts/prepare_alpaca.py:21]: Can you check how many samples does alpaca-lora have in total? We should set the percentage that corresponds to a test split size of 2k to keep the same behaviour

File: scripts/prepare_alpaca.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def prepare(
     tokenizer = Tokenizer(checkpoint_dir)
 
     # Partition the dataset into train and test
+    test_split_size = len(data) * test_split_percent // 100
```
Comment (carmocca) [inline scripts/prepare_alpaca.py:57]: `random_split` supports having lengths as percentages: https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split
which means that we can simplify this further

File: scripts/prepare_alpaca.py (hunk 1 above)
Comment (patrickhwood) [reply scripts/prepare_alpaca.py:21]: Sure.  There are 51759 samples, so 4% would give us 2070 in the test set.

Comment (patrickhwood) [reply scripts/prepare_alpaca.py:21]: Just a side comment: seems like 2K samples out of 52K is rather small for the test set, no?  I'm just starting to work on training LLMs, but for image detection/segmentation, 10-20% seems to be the sweet spot.

File: scripts/prepare_alpaca.py (hunk 2 above)
Comment (patrickhwood) [reply scripts/prepare_alpaca.py:57]: Cool!  I didn't know that.

File: scripts/prepare_alpaca.py (hunk 1 above)
Comment (patrickhwood) [reply scripts/prepare_alpaca.py]: Done.  test_split_percent is now a float, and the default produces exactly 2000 test samples.

File: scripts/prepare_alpaca.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 DATA_FILE_NAME = "alpaca_data_cleaned_archive.json"
 DESTINATION_PATH = Path("data/alpaca")
 CHECKPOINT_DIR = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
-TEST_SPLIT_SIZE = 2000
+TEST_SPLIT_PERCENT = 3.865	# to get exactly 2000 test samples
```
Comment (carmocca) [inline scripts/prepare_alpaca.py:21]: It's more common to represent a percentage as a fraction of 1. This will simplify the code below because you won't need to divide by 100

Also avoiding tabs

```suggestion
TEST_SPLIT_PERCENT = 0.3865  # percentage to get exactly 2000 samples
```

File: scripts/prepare_alpaca.py [hunk 4]
```diff
@@ -54,9 +54,9 @@ def prepare(
     tokenizer = Tokenizer(checkpoint_dir)
 
     # Partition the dataset into train and test
-    train_split_size = len(data) - test_split_size
+    train_split_percent = 100 - test_split_percent
```
Comment (carmocca) [inline scripts/prepare_alpaca.py:57]:
```suggestion
    train_split_percent = 1.0 - test_split_percent
```

File: scripts/prepare_alpaca.py [hunk 5]
```diff
@@ -54,9 +54,9 @@ def prepare(
...
     train_set, test_set = random_split(
-        data, lengths=(train_split_size, test_split_size), generator=torch.Generator().manual_seed(seed)
+        data, [train_split_percent / 100., test_split_percent / 100.], generator=torch.Generator().manual_seed(seed)
```
Comment (carmocca) [inline scripts/prepare_alpaca.py:59]:
```suggestion
        data, (train_split_percent, test_split_percent), generator=torch.Generator().manual_seed(seed)
```

**Review by carmocca — APPROVED** [review]
Thank you!

File: scripts/prepare_alpaca.py (hunk 4 above)
Comment (patrickhwood) [reply scripts/prepare_alpaca.py:57]: Yes, changing it to a fraction makes sense, but the variable names should also be changed, as they are no longer percentages.  Percent comes from the Latin phrase "per centum" which means "by the hundred" or "for each hundred."

File: scripts/prepare_alpaca.py (hunk 3 above)
Comment (patrickhwood) [reply scripts/prepare_alpaca.py:21]: Needs to be 0.03865.

File: scripts/prepare_alpaca.py (hunk 5 above)
Comment (patrickhwood) [reply scripts/prepare_alpaca.py]: Why a tuple instead of a list?  The examples at https://pytorch.org/docs/stable/data.html are lists.  I suppose either will work, but I prefer to follow any usage shown in the documentation.

Comment (carmocca) [reply scripts/prepare_alpaca.py]: It's a very minor nit. But if you don't need your sequence to be mutable, it's faster to use tuples
